e2e/test_operator_upgrades/operator_upgrades_test.go

```diff
@@ -874,4 +967,52 @@ var _ = Describe("Operator Upgrades", Label("e2e"), func() {
 		EntryDescription("Upgrade from %[1]s to %[2]s and one process is missing the new binary"),
 		fixtures.GenerateUpgradeTableEntries(testOptions),
 	)
+
+	DescribeTable(
+		"upgrading a cluster when no storage processes are restarted",
+		func(beforeVersion string, targetVersion string) {
+			// We set the before version here to overwrite the before version from the specific flag
+			// the specific flag will be removed in the future.
+			isAtLeast := factory.OperatorIsAtLeast(
+				"v1.14.0",
+			)
+
+			if !isAtLeast {
+				Skip("operator doesn't support feature for test case")
+			}
+
+			clusterSetup(beforeVersion, true)
+
+			// Select storage processes and use the buggify option to skip those
+			// processes during the restart command.
+			storagePods := fdbCluster.GetStoragePods()
+			Expect(storagePods.Items).NotTo(BeEmpty())
+
+			ignoreDuringRestart := make(
+				[]fdbv1beta2.ProcessGroupID,
+				0,
+				len(storagePods.Items),
+			)
+
+			for _, pod := range storagePods.Items {
+				ignoreDuringRestart = append(
+					ignoreDuringRestart,
+					fdbv1beta2.ProcessGroupID(pod.Labels[fdbCluster.GetCachedCluster().GetProcessGroupIDLabel()]),
+				)
+			}
+
+			log.Println(
+				"Selected Pods:",
+				ignoreDuringRestart,
+				"to be skipped during the restart",
+			)
+			fdbCluster.SetIgnoreDuringRestart(ignoreDuringRestart)
+
+			// The cluster should still be able to upgrade.
+			Expect(fdbCluster.UpgradeCluster(targetVersion, true)).NotTo(HaveOccurred())
```

Comment:
Are you sure the upgrade will succeed if no storage server is upgraded? That would probably result in "dataloss" and that should block the operator from doing further actions.

Reply:
The output of "kubectl-fdb analyze" over the cluster after running the test: https://github.com/FoundationDB/fdb-kubernetes-operator/pull/1580#issuecomment-1501916726

The test did throw an error though (which I mentioned in https://github.com/FoundationDB/fdb-kubernetes-operator/pull/1580#issuecomment-1501916726).

I don't have any more insights into this that what is reported by "kubectl-fdb analyze".

Maybe we should have a theory as to what is supposed to happen in this scenario and verify whether the test output matches that theory or not (rather than simply running the test and observe the outcome, which we are trying to do here).

Reply:
I ran this test again. (The behavior is similar to what I reported here: https://github.com/FoundationDB/fdb-kubernetes-operator/pull/1580#issuecomment-1501916726).

The test failed with this error:

> 
> <*errors.errorString | 0xc002b67680>: {
>           s: "invariant InvariantClusterStatusAvailableWithThreshold failed",
>       }
>       invariant InvariantClusterStatusAvailableWithThreshold failed
>   occurred
> 

Output of "kubectl-fdb analyze":

```
# kubectl-fdb analyze fdb-cluster-in36toff -n  sre-nawokdnf
Checking cluster: sre-nawokdnf/fdb-cluster-in36toff
✔ Cluster is available
✔ Cluster is fully replicated
✔ Cluster is reconciled
✔ ProcessGroups are all in ready condition
✔ Pods are all running and available
Checking cluster: sre-nawokdnf/fdb-cluster-in36toff with auto-fix: false
```

The output of "kubectl -n  sre-nawokdnf get fdb,po" shows that the storage processes are running at 7.1.27 version (I checked the version of fdbserver binary in one of the storage pods).

Reply:
The analyze command is mostly reading the information from the FoundationDB cluster object and checking the Pods. We should check the operator logs to get a better understanding of the behaviour.

Reply:
Removed this test from this PR. Will make another PR for that test.